cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/UserClient.java

```diff
@@ -422,4 +421,18 @@ protected RegistryClient getRegistryClient()
         return new RegistryClient();
     }
 
+    private AuthMethod getAuthMethod()
+    {
+        Subject subject = AuthenticationUtil.getCurrentSubject();
+        for (Object o : subject.getPublicCredentials())
+        {
+            if (o instanceof X509CertificateChain)
+                return AuthMethod.CERT;
+            if (o instanceof SSOCookieCredential)
+                return AuthMethod.COOKIE;
+            // AuthMethod.PASSWORD not supported
+            // AuthMethod.TOKEN not supported
+        }
+        return AuthMethod.ANON;
```

Comment:
getUser and getDisplayUsers must be authenticated calls.  So, instead of waiting for the server return the 403 you may as well fail with a AccessControlException here.